### PR TITLE
WINDUP-4174: Release Notes for MTR 1.2.6

### PR DESCRIPTION
**Pull request by anarnold97**
### JIRA

* [WINDUP-4174](https://issues.redhat.com/browse/WINDUP-4174)

### PREVIEW

* [MTR 1.2.6 release notes](https://deploy-preview-877--windup-documentation.netlify.app/docs/release-notes-mtr/master/#mtr-1-2-6)

    * [Known issues](https://deploy-preview-877--windup-documentation.netlify.app/docs/release-notes-mtr/master/#mtr-rn-known-issues-1-2-6_release-notes)
    * [Resolved issues](https://deploy-preview-877--windup-documentation.netlify.app/docs/release-notes-mtr/master/#mtr-rn-resolved-issues-1-2-6_release-notes)

### CODE REVIEW

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//
+// * docs/release-notes-mtr/mtr_release_notes-1.2/master.adoc
+
+:_content-type: REFERENCE
+[id="mtr-rn-resolved-issues-1-2-6_{context}"]
+= Resolved issues
+
+{ProductName} ({ProductShortName}) 1.2.6 resolves the following issues:
+
+.CVE-2023-45857: Axios 1.5	exposes confidential data stored in cookies
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:11]:
```suggestion
.CVE-2023-45857: Axios 1.5 exposes confidential data stored in cookies
```

File: docs/topics/mtr-rn-new-features-1-2-6.adoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Module included in the following assemblies:
+//
+// * docs/release_notes/master.adoc
+
+:_content-type: CONCEPT
+[id="rn-new-features-1-2-6_{context}"]
+= New features
+
+// This section describes the new features of the {ProductName} ({ProductShortName}) 1.2.5:
```
Comment (anarnold97) [inline docs/topics/mtr-rn-new-features-1-2-6.adoc:9]:
```suggestion
```

File: docs/topics/mtr-rn-new-features-1-2-6.adoc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Module included in the following assemblies:
+//
+// * docs/release_notes/master.adoc
+
+:_content-type: CONCEPT
+[id="rn-new-features-1-2-6_{context}"]
+= New features
+
+
```
Comment (anarnold97) [inline docs/topics/mtr-rn-new-features-1-2-6.adoc:9]:
```suggestion
```

File: docs/release-notes-mtr/master.adoc [hunk 4]
```diff
@@ -17,6 +17,11 @@ include::topics/making-open-source-more-inclusive.adoc[]
 
 These release notes cover all Z-stream releases of {ProductShortName} 1.2 with the most recent release listed first.
 
+== {ProductShortName} 1.2.6
+include::topics/mtr-rn-new-features-1-2-6.adoc[leveloffset=+2]
```
Comment (anarnold97) [inline docs/release-notes-mtr/master.adoc:21]:
```suggestion
// include::topics/mtr-rn-new-features-1-2-6.adoc[leveloffset=+2]
```

File: docs/topics/mtr-rn-known-issues-1-2-6.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// * docs/release-notes-mtr/master.adoc
+
+:_content-type: REFERENCE
+[id="mtr-rn-known-issues-1-2-6_{context}"]
+
+= Known issues
+
+There are no major known issues in this {ProductName} ({ProductShortName}) 1.2.6 release.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-known-issues-1-2-6.adoc:10]:
```suggestion
There are no major known issues in the {ProductShortName} 1.2.6 release.
```

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc [hunk 6]
```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// * docs/release-notes-mtr/mtr_release_notes-1.2/master.adoc
+
+:_content-type: REFERENCE
+[id="mtr-rn-resolved-issues-1-2-6_{context}"]
+= Resolved issues
+
+{ProductName} ({ProductShortName}) 1.2.6 resolves the following issues:
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:9]:
```suggestion
{ProductShortName} 1.2.6 has the following resolved issues:
```

File: docs/topics/mtr-rn-known-issues-1-2-6.adoc [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+// * docs/release-notes-mtr/master.adoc
+
+:_content-type: REFERENCE
+[id="mtr-rn-known-issues-1-2-6_{context}"]
+
+= Known issues
+
+There are no major known issues in the {ProductShortName} 1.2.6 release.
+
+For a complete list of all known issues, see the list of link:https://issues.redhat.com/issues/?filter=12435316[MTR 1.2.6 known issues] in Jira.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-known-issues-1-2-6.adoc:12]:
```suggestion
For a complete list of all known issues, see the list of link:https://issues.redhat.com/issues/?filter=12436484[MTR 1.2.6 known issues] in Jira.
```

File: docs/topics/mtr-rn-known-issues-1-2-6.adoc [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+:_content-type: REFERENCE
+[id="mtr-rn-known-issues-1-2-6_{context}"]
+
+= Known issues
+
+The following known issues are in the {ProductShortName} 1.2.6 release:
+
+.Unable to migrate application to {ProductShortName} due to a `SEVERE [org.jboss.windup.web.services.messaging.PackageDiscoveryMDB]` error
+
+When uploading files for analyze, the server log would print a `SEREVE` message, and the webpage could freeze at the discovering and fetching packages stage, returning an `SEVERE [org.jboss.windup.web.services.messaging.PackageDiscoveryMDB]` error. This error is caused by a `null: java.lang.NullPointerException`. link:https://issues.redhat.com/browse/WINDUP-4189[(WINDUP-4189)]
```
Comment (anarnold97) [inline docs/topics/mtr-rn-known-issues-1-2-6.adoc:14]:
```suggestion
When uploading files for analyze, the server log would return a `SEVERE [org.jboss.windup.web.services.messaging.PackageDiscoveryMDB]` error. This error is caused by a `null: java.lang.NullPointerException`. link:https://issues.redhat.com/browse/WINDUP-4189[(WINDUP-4189)]
```

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+:_content-type: REFERENCE
+[id="mtr-rn-resolved-issues-1-2-6_{context}"]
+= Resolved issues
+
+{ProductShortName} 1.2.6 has the following resolved issues:
+
+.CVE-2023-45857: Axios 1.5 exposes confidential data stored in cookies
+
+A flaw was discovered in Axios 1.5.1 that accidentally revealed the confidential `XSRF-TOKEN` stored in cookies by including it in the HTTP header `X-XSRF-TOKEN` for every request made to any host, thereby allowing attackers to view sensitive information. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:19]:
```suggestion
A flaw was discovered in Axios 1.5.1 that accidentally revealed the confidential `XSRF-TOKEN`, stored in cookies, by including it in the HTTP header `X-XSRF-TOKEN` for every request made to any host, thereby allowing attackers to view sensitive information. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+.CVE-2024-28849: `follow-redirects` package clears authorization headers
+
+A flaw was discovered in the `follow-redirects` package, which clears authorization headers, but it fails to clear the `proxy-authentication` headers. This flaw could lead to credential leakage, which could have a high impact on data confidentiality.
+Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-28849[(CVE-2024-28849)]
+
+.CVE-2024-29131: Out-of-bounds Write vulnerability in Apache Commons Configuration
+
+A vulnerability was found in Apache Commons-Configuration2, where a Stack Overflow Error can occur when adding a property in AbstractListDelimiterHandler.flattenIterator(). This issue could allow an attacker to corrupt memory or execute a denial of service attack by crafting malicious property that triggers an out-of-bounds write issue when processed by the vulnerable method. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:33]:
```suggestion
A vulnerability was found in Apache Commons-Configuration2, where a Stack Overflow Error can occur when adding a property in the `AbstractListDelimiterHandler.flattenIterator()` method. This issue could allow an attacker to corrupt memory or execute a denial of service (DoS) attack by crafting a malicious property that triggers an out-of-bounds write issue when processed by the vulnerable method. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+.CVE-2024-29131: Out-of-bounds Write vulnerability in Apache Commons Configuration
+
+A vulnerability was found in Apache Commons-Configuration2, where a Stack Overflow Error can occur when adding a property in the `AbstractListDelimiterHandler.flattenIterator()` method. This issue could allow an attacker to corrupt memory or execute a denial of service (DoS) attack by crafting a malicious property that triggers an out-of-bounds write issue when processed by the vulnerable method. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-29131[(CVE-2024-29131)]
+
+.CVE-2024-29133: Out-of-bounds Write vulnerability in Apache Commons Configuration
+
+A vulnerability was found in Apache Commons-Configuration2, where a Stack Overflow Error occurs when calling ListDelimiterHandler.flatten(Object, int) with a cyclical object tree. This issue could allow an attacker to trigger an out-of-bounds write that could lead to memory corruption or cause a denial of service condition. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:39]:
```suggestion
A vulnerability was found in Apache Commons-Configuration2, where a Stack Overflow Error occurs when calling the `ListDelimiterHandler.flatten(Object, int)` method with a cyclical object tree. This issue could allow an attacker to trigger an out-of-bounds write that could lead to memory corruption or cause a denial of service (DoS) attach. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+A vulnerability was found in Apache Commons-Configuration2, where a Stack Overflow Error occurs when calling the `ListDelimiterHandler.flatten(Object, int)` method with a cyclical object tree. This issue could allow an attacker to trigger an out-of-bounds write that could lead to memory corruption or cause a denial of service (DoS) attach. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-29133[(CVE-2024-29133)]
+
+.CVE-2024-29180: `webpack-dev-middleware` lack of URL validation may lead to file leak
+
+A flaw was found in the webpack-dev-middleware package, where it failed to validate the supplied URL address sufficiently before returning local files. This flaw allows an attacker to craft URLs to return arbitrary local files from the developer's machine. The lack of normalization before calling the middleware also allows the attacker to perform path traversal attacks on the target environment. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:45]:
```suggestion
A flaw was found in the `webpack-dev-middleware` package, where it failed to validate the supplied URL address sufficiently before returning local files. This flaw allows an attacker to craft URLs to return arbitrary local files from the developer's machine. The lack of normalization before calling the middleware also allows the attacker to perform path traversal attacks on the target environment. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
```

**Review by kpunwatk — APPROVED** [review]
LGTM!

**Comment by PhilipCattanach** [comment]
@anarnold97 They look good to me. Please publish.
Sorry for the delay in responding.

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-29133[(CVE-2024-29133)]
+
+.CVE-2024-29180: `webpack-dev-middleware` lack of URL validation may lead to file leak
+
+A flaw was found in the `webpack-dev-middleware` package, where it failed to validate the supplied URL address sufficiently before returning local files. This flaw allows an attacker to craft URLs to return arbitrary local files from the developer's machine. The lack of normalization before calling the middleware also allows the attacker to perform path traversal attacks on the target environment. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-29180[(CVE-2024-29180)]
+
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:48]:
```suggestion

.CVE-2023-4639: `org.keycloak-keycloak-parent` undertow Cookie Smuggling and Spoofing 

A flaw was found in Undertow, which incorrectly parses cookies with certain value-delimiting characters in incoming requests. This vulnerability has the potential to enable an attacker to construct a cookie value to intercept `HttpOnly` cookie values or spoof arbitrary additional cookie values, resulting in unauthorized data access or modification. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.

For more details, see link:https://access.redhat.com/security/cve/CVE-2023-4639[(CVE-2023-4639)].


.CVE-2023-36479: `com.google.guava-guava-parent` improper addition of quotation marks to user inputs in Jetty CGI Servlet 

A flaw was found in Jetty's `org.eclipse.jetty.servlets.CGI` Servlet, which permits incorrect command execution in specific circumstances, such as requests with certain characters in requested filenames. This issue could allow an attacker to run permitted commands besides the ones requested. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.

For more details, see link:https://access.redhat.com/security/cve/CVE-2023-36479[(CVE-2023-36479)].


.CVE-2023-26364: `css-tools` improper input validation causes denial of service


A flaw was found in `@adobe/css-tools`, which could potentially lead to a minor denial of service (DoS) when parsing CSS. User interaction and privileges are not required to jeopardize an environment. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.

For more details, see link:https://access.redhat.com/security/cve/CVE-2023-26364[(CVE-2023-26364)].


.CVE-2023-48631: `css-tools`: regular expression denial of service

A flaw was found in `@adobe/css-tools`, which could lead to a regular expression denial of service (ReDoS) when attempting to parse CSS. Users are recommended to upgrade to {ProductShortName} 1.2.6, which resolves this issue.

For more details, see link:https://access.redhat.com/security/cve/CVE-2023-48631[(CVE-2023-48631)].


```

Comment (anarnold97) [reply docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:48]: @mrizzi && @kpunwatk && @PhilipCattanach — added the CVEs i missed

Is @raoradhika ok to merge and publish, after I commit?

Comment (mrizzi) [reply docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:48]: @anarnold97 thanks :+1:

**Review by mrizzi — APPROVED** [review]

File: docs/topics/mtr-rn-resolved-issues-1-2-6.adoc (hunk 13 above)
Comment (kpunwatk) [reply docs/topics/mtr-rn-resolved-issues-1-2-6.adoc:48]: LGTM, Thank you @anarnold97!

**Pull request merged**